engine/src/main/java/org/destinationsol/components/HealthComponent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.components;
+
+import org.terasology.gestalt.entitysystem.component.Component;
+
+public final class HealthComponent implements Component<HealthComponent> {
```

Comment:
Gestalt 7 doesn't require component class names to end in `Component` so I would just call this `Health`

Reply:
I'm working on the docs now. I can't believe I forgot XD
I'd rather use `isPresent()`, but I'm ok with either. Is either one preferable for any reason?
In terms of naming, should Events also not be called `ExampleEvent`?

Reply:
`isPresent()` is fine by me. `ifPresent()` is nice in that you don't need to `.get()` the component but `isPresent()` is slightly more readable.
You raise a very good point about the naming. I was thinking that I don't really want have to put `HealthComponent` into the prefabs. Not really a huge deal so I'll let you decide :)

Reply:
I think that it should be `Health` and `DamageEvent`. With components, the essence is really the thing itself, e.g. health, and `component` is a comment about the software architecture. With events, the main thing is that an event is happening. The "`Damage`" is just data about the event.